### DISCUSSION TITLE
earthaccess @ AGU  🛰️

### DISCUSSION THREAD

**Discussion by betolink** [earthaccess-dev/earthaccess#337] [General]
Great news! `earthaccess` got a spot at the NASA demo booth! Thanks to @asteiker for submitting the application!

<img src="https://www.earthdata.nasa.gov/s3fs-public/2022-12/agu2021_smd-tables_art.jpeg" alt="NASA @ AGU" align="right"  width="400"/>

From NASA:

> The demo area will have a monitor and podium. It offers a more intimate space for showcasing tools and data sets and having Q&A. 
> Demos will be promoted alongside Hyperwall talks on the NASA Booth schedule.
> Timeslots at the demo area are longer than the Hyperwall timeslots. While we do not currently have the final minute count, we’re 
> looking at about 30 minutes.

Now the question is, what should we demo in 30 minutes? 
Some topics that come to mind are:

* Issues earthaccess addresses
* `earthaccess` as a tool for in-place analysis in the cloud
* Reproducible workflows with earthaccess 
* earthaccess at scale with Dask 

I guess most people at AGU won't know anything about this library or why it's useful but most of them know about remote sensing data from NASA. What other ideas would be interesting to showcase? 

For reference I'm attaching the poster I presented at AGU last year. 
[earthaccess AGU draft.pdf](https://github.com/nsidc/earthaccess/files/13230502/earthaccess.AGU.draft.pdf)

**Comment by jrbourbeau** [comment]
Great news indeed! 

From talking with a few groups, I think the message that applies most broadly is that `earthaccess` handles authentication and retrieving the data files you want from NASA Earthdata in a simple way. Then after you've gotten your data files, `earthaccess` steps into the background and you can use the normal PyData stack (e.g. `xarray`, `numpy`, `h5py`, `scipy`, etc) like you're used to. 

I've seen lots of notebooks with 10-30 lines of complicated `requests` / `s3fs` / ... code that handles Earthdata authentication + data file retrieval. Being able to replace that with something simple like:

```python
import earthaccess

results = earthaccess.search_data(
    short_name="MUR-JPL-L4-GLOB-v4.1",
    temporal=("2020-01-01", "2021-12-31"),
)
```

really resonates with folks. 

I also like the idea of showing something at scale. If we had, for example, 20 lines of code resulted in us churning through several GBs or TBs of data, with the final output being a nice plot of some meaningful quantity for the AGU / NASA crowd, I think that'd make for a nice demo.
